Addition of Add new card CTA

oikeme opened the pull request:
Add the "Add new card CTA" to the payments method page for Jetpack Manage

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Closes https://github.com/Automattic/jetpack-genesis/issues/189


## Proposed Changes

* Update the Payments Method page to use the partner portal styles and add the new "Add new card" CTA in the top right.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Use the Jetpack cloud live link below and go to the new Payment Methods page (/partner-portal/payment-methods/add?flags=jetpack/card-addition-improvements)
* Verify that the styles for the header are now consistent with the other pages in the Purchases
* Verify that the "Add new card" CTA is visible
* Verify that the page is responsive and works as expected for different screen sizes


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

oikeme committed:
Addition of Add new card CTA

Add the "Add new card CTA" to the payments method page for Jetpack Manage

yashwin commented on client/jetpack-cloud/sections/partner-portal/payment-methods-v2/index.tsx:
```diff
@@ -1,10 +1,11 @@
-	LayoutHeaderTitle as Title,
-} from 'calypso/jetpack-cloud/components/layout/header';
-import LayoutTop from 'calypso/jetpack-cloud/components/layout/top';
+import CardHeading from 'calypso/components/card-heading';
+import Layout from '../layout';
+import LayoutHeader from '../layout/header';
+import LayoutTop from '../layout/top';
```
I suggest using the layout-related component from `jetpack-cloud/components` since it was designed to be reused throughout the Jetpack Cloud.

oikeme replied:
Thanks for the heads up. I was reviewing the other `Purchases` pages and noticed the previous implementations so I went with `partner-portal` for consistency but this change makes sense.

yashwin commented on client/jetpack-cloud/sections/partner-portal/payment-methods-v2/index.tsx:
```diff
@@ -18,9 +19,12 @@ export default function PaymentMethodListV2() {
-					<Title>{ title } </Title>
-					<Subtitle>{ subtitle }</Subtitle>
+					<CardHeading size={ 36 }>{ title }</CardHeading>
+					<Button href="/partner-portal/payment-methods/add" primary>
+						{ translate( 'Add new card' ) }
+					</Button>
```
We could use `LayoutHeaderActions` to handle this. You can use this [file](https://github.com/Automattic/wp-calypso/blob/db5e20798888f32a080aeaa292ae607b0199a1e5/client/jetpack-cloud/sections/partner-portal/issue-license-v2/index.tsx) as reference

yashwin reviewed (commented):
Thanks for working on this, @oikeme! 

It works as expected. However, I have left a few comments.

oikeme committed:
Update Add new payments page

Update the Add new payment methods page to to use previous jetpack-cloud component

oikeme committed:
Delete styles.scss

Remove unused css
